_posts/en/newsletters/2023-05-17-newsletter.md

```diff
@@ -0,0 +1,242 @@
+---
+title: 'Bitcoin Optech Newsletter #251'
+permalink: /en/newsletters/2023/05/17/
+name: 2023-05-17-newsletter
+slug: 2023-05-17-newsletter
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter describes a proposal to begin testing HTLC
+endorsement, relays a request for feedback about proposed specifications
+for Lightning Service Providers (LSPs), discusses challenges with
+opening zero-conf channels when using dual funding, looks at a
+suggestion for advanced applications of payjoin transactions, and links
+to summaries of a recent in-person meeting of Bitcoin Core developers.
+Also included in this week's newsletter is the first part of a new
+series about policies for transaction relay and mempool inclusion, plus
+our regular sections announcing new releases and release candidates and
+describing notable changes to popular Bitcoin infrastructure software.
+
+## News
+
+- **Testing HTLC endorsement:** several weeks ago, Carla Kirk-Cohen and
+  Clara Shikhelman [posted][kcs endorsement] to the Lightning-Dev mailing list about
+  the next steps they and others planned to take to test the idea of
+  [HTLC][topic htlc] endorsement (see [Newsletter #239][news239
+  endorsement]) as part of a mitigation for [channel jamming
+  attacks][topic channel jamming attacks].  Most notably, they provided a
+  short [proposed specification][bolts #1071] that could be deployed
+  using an experimental flag, preventing deployments of it from having any effect on
+  interactions with non-participating nodes.
+
+    Once deployed by experimenters, it should become easier to answer
+    one of the [constructive criticisms][decker endorsement] of this idea, which is how
+    many payments originate from spenders who frequently use the same
+    peers and roughly the same routes.  If the core users of LN are
+    frequently sending payments to each other over many of the same
+    routes, and if the reputation system works as planned, then that
+    core network will be more likely to keep functioning during a
+    channel jamming attack.  But if most spenders only send payments
+    rarely (or only send their most critical types of payments rarely,
+    such as high-value payments), then they won't have enough
+    interactions to build a reputation, or the reputation data will lag
+    far behind the current state of the network (making it less useful
+    or even allowing reputation to be abused).
+
```

Comment:
A reply email [0] was sent today. In the email, we write that the number of spenders that use the same route is not the important data point, as nodes only track the reputation of their neighbors. We do recognize that during an attack, a small node will have a difficult time routing through a big node. In times of peace, on the other hand, the small node would not be affected by the new scheme.

[0] https://lists.linuxfoundation.org/pipermail/lightning-dev/2023-May/003958.html

Reply:
Thanks for the link!  I plan on making a slight change to partly address your point:

```diff
     Once deployed by experimenters, it should become easier to answer
     one of the [constructive criticisms][decker endorsement] of this idea, which is how
-    many payments originate from spenders who frequently use the same
-    peers and roughly the same routes.  If the core users of LN are
+    many forwarded payments would actually receive a boost from the
+    scheme.  If the core users of LN are
     frequently sending payments to each other over many of the same
     routes, and if the reputation system works as planned, then that
     core network will be more likely to keep functioning during a
```

I don't want to over-discuss the protocol here (I'm happy to engage on list or to join one of the calls), but where Decker's comments resonate with me is the idea that there might not be enough repeat engagement to build a significant reputation.  As I understand your protocol, it's deliberately designed to be fragile in order to prevent abuse.  Spender Alice must endorse her own HTLC in the first hop; Bob must be confident enough in Alice to endorse the HTLC in the second hop to Carol; Carol must be confident in Bob on the hop to Dan; etc...  If at any stage, an HTLC isn't endorsed, it will remain unendorsed for the remainder of the hops.

If Alice only sends a few payments a month, what basis will Bob have for endorsing Alice, especially if he risks his relationship with Carol (from whom he might receive many multiples of the forwarding income he receives from Alice's occasional HTLCs)?  Same if Bob only rarely forwards to Carol.

If a very large percentage of LN payments originate from infrequent spenders or cross rarely used forwarding nodes, then it seems possible that the protocol won't work (unless I'm misunderstanding something!).  I don't actually think that's the case---I would guess that there are plenty enough frequent spenders and highly used forwarders for it to work now, and that the mean frequency of spending per node will probably increase over time, but I felt that it's worth highlighting Decker's critique (or at least my perception of it) in advance of experimental data being received.

Reply:
Thanks for your comment, and I agree that Decker's critique is worth mentioning. Looking forward to discussing this further in the ml or calls!